Sources/MapboxMaps/Foundation/MapView.swift

```diff
@@ -326,6 +335,74 @@ open class MapView: UIView {
             mapboxMap: mapboxMap)
     }
 
+    private func subscribeToLifecycleNotifications() {
+        if bundle.infoDictionary?["UIApplicationSceneManifest"] != nil {
+            if #available(iOS 13.0, *) {
+                notificationCenter.addObserver(self,
+                                               selector: #selector(didReceiveNotification(_:)),
+                                               name: UIScene.willEnterForegroundNotification,
+                                               object: window?.parentScene)
+                notificationCenter.addObserver(self,
+                                               selector: #selector(didReceiveNotification(_:)),
+                                               name: UIScene.didEnterBackgroundNotification,
+                                               object: window?.parentScene)
+            } else {
+                notificationCenter.addObserver(self,
+                                               selector: #selector(didReceiveNotification(_:)),
+                                               name: UIApplication.willEnterForegroundNotification,
+                                               object: nil)
+                notificationCenter.addObserver(self,
+                                               selector: #selector(didReceiveNotification(_:)),
+                                               name: UIApplication.didEnterBackgroundNotification,
+                                               object: nil)
+            }
+        } else {
+            notificationCenter.addObserver(self,
+                                           selector: #selector(didReceiveNotification(_:)),
+                                           name: UIApplication.willEnterForegroundNotification,
+                                           object: nil)
+            notificationCenter.addObserver(self,
+                                           selector: #selector(didReceiveNotification(_:)),
+                                           name: UIApplication.didEnterBackgroundNotification,
+                                           object: nil)
+        }
+    }
+
+    private func unsubscribeFromLifecycleNotifications() {
+        if #available(iOS 13.0, *) {
+            notificationCenter.removeObserver(self, name: UIScene.willEnterForegroundNotification, object: nil)
+            notificationCenter.removeObserver(self, name: UIScene.didEnterBackgroundNotification, object: nil)
+        }
+        notificationCenter.removeObserver(self, name: UIApplication.willEnterForegroundNotification, object: nil)
+        notificationCenter.removeObserver(self, name: UIApplication.didEnterBackgroundNotification, object: nil)
+    }
+
+    @objc private func didReceiveNotification(_ notification: Notification) {
```

Comment:
Can we name this more specifically?

Reply:
Hmm, I guess I'm saying I'd rather have separate 3 separate handlers for foreground, background, and memory warnings.